massa-execution/src/interface_impl.rs

```diff
@@ -155,4 +158,22 @@ impl Interface for InterfaceImpl {
             .map(|addr| addr.to_bs58_check())
             .collect())
     }
+
+    fn generate_event(&self, data: String) -> Result<()> {
+        let slot = context_guard!(self).slot;
```

Comment:
let's not use context_guard! at every line. This causes CPU-wide sync every time and is very heavy. Better lock once at the beginning of the function (see how it was implemented in other methods of InterfaceImpl)